src/services/codefixes/inferFromUsage.ts

```diff
@@ -665,29 +667,83 @@ namespace ts.codefix {
             }
         }
 
-        function getTypeFromUsageContext(usageContext: UsageContext, checker: TypeChecker): Type | undefined {
-            if (usageContext.isNumberOrString && !usageContext.isNumber && !usageContext.isString) {
-                return checker.getUnionType([checker.getNumberType(), checker.getStringType()]);
+        function unifyFromContext(inferences: Type[], checker: TypeChecker, fallback = checker.getAnyType()): Type {
+            inferences.push(fallback);

```

Comment:
This passes all tests if I replace this with `if (!inferences.length) return fallback;`, which means `inferences` could be a `ReadonlyArray`.

Reply:
Makes sense to keep it this way until a counter-example shows up.